src/services/accounts/AccountsStakingPayoutsService.ts

```diff
@@ -513,9 +529,26 @@ export class AccountsStakingPayoutsService extends AbstractService {
 				return {
 					commission,
 				};
+			} else if (stakingVersion < 14) {
```

Comment:
I made some changes and reduced the use of the `stakingVersion`. The bug you mention was a few lines below, but I changed it so that if the `stakingVersion` is below 14, it goes with the old logic, but if it's not, instead of replacing `validatorLedger.legacyClaimedRewards` with `claimedRewards`, the value of the era gets added to the array.